### PR TITLE
gps-ubx: retrieve accuracy values from messages.

### PR DESCRIPTION
**Pull request by mlyle**
Should get plumbed down to INS13 variances eventually...

### CODE REVIEW

File: shared/uavobjectdefinition/gpsposition.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
         <field name="Heading" units="degrees" type="float" elements="1"/>
         <field name="Groundspeed" units="m/s" type="float" elements="1"/>
         <field name="Satellites" units="" type="int8" elements="1"/>
+        <field name="Accuracy" units="" type="float" elements="1"/>
```
Comment (peabody124) [inline shared/uavobjectdefinition/gpsposition.xml]: no units?

Comment (peabody124) [inline shared/uavobjectdefinition/gpsposition.xml]: looks like cm from the notes

File: flight/Modules/GPS/NMEA.c [hunk 2]
```diff
@@ -28,6 +28,8 @@
  * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
+#include <math.h>
```
Comment (peabody124) [inline flight/Modules/GPS/NMEA.c]: this was needed to do multiplication?

File: flight/Modules/GPS/UBX.c [hunk 3]
```diff
@@ -251,6 +252,7 @@ static void parse_ubx_nav_velned (const struct UBX_NAV_VELNED *velned, GPSPositi
 			GpsVelocity.North	= (float)velned->velN/100.0f;
 			GpsVelocity.East	= (float)velned->velE/100.0f;
 			GpsVelocity.Down	= (float)velned->velD/100.0f;
+			GpsVelocity.Accuracy	= (float)velned->sAcc/100.0f;
```
Comment (peabody124) [inline flight/Modules/GPS/UBX.c]: according to ```struct UBX_NAV_SOL``` this starts at cm/s accuracy. you are trying to convert to 100µm/s ?

**Comment by peabody124** [comment]
This will be really useful - to see if this drops when we see issues, even if PDOP or Sat doesn't.

File: flight/Modules/GPS/NMEA.c (hunk 2 above)
Comment (mlyle) [inline flight/Modules/GPS/NMEA.c]: Woops.  I wavered with whether I should set it to NAN, so I picked up that line.  Willfix.

File: flight/Modules/GPS/UBX.c (hunk 3 above)
Comment (mlyle) [inline flight/Modules/GPS/UBX.c]: cm/s * 1m / 100cm = m/s, just like the other terms.

**Comment by mlyle** [comment]
Rewrote history based on comments.

**Comment by elafargue** [comment]
Ideally, the accuracy info should also be added to the GPS Display gadget

File: flight/Modules/GPS/UBX.c (hunk 3 above)
Comment (peabody124) [inline flight/Modules/GPS/UBX.c]: don't tell anyone i did that ... :-p

**Comment by peabody124** [comment]
Looks good. Hopefully we get some logs with this info for study. I'll merge the firmware changes and we can have a separate issue for the GCS side.

File: flight/Modules/GPS/UBX.c (hunk 3 above)
Comment (mlyle) [inline flight/Modules/GPS/UBX.c]: i want forgiveness/the same immunity next time i do ;)

**Pull request merged**